### PR TITLE
[#6791] allow both regex- and glob-style matching (main)

### PR DESCRIPTION
**Pull request by d-w-moore**
… matching

### CODE REVIEW

**Comment by d-w-moore** [comment]
Still testing.

**Review by korydraughn — COMMENTED** [review]
Changes are good.

Awaiting test results.

**Comment by alanking** [comment]
Cherry-picked from https://github.com/irods/irods_client_icommands/pull/340

File: src/itree.cpp
Comment (d-w-moore) [reply src/itree.cpp]: Right, not needed.  Now amended.

**Comment by d-w-moore** [comment]
All tests pass, so we're good to merge this and irods/irods#6945 pending approval

File: src/itree.cpp
Comment (alanking) [inline src/itree.cpp:42]: Extra line

**Review by alanking — COMMENTED** [review]
Should we match the commit message to the one in 4-2-stable? https://github.com/irods/irods_client_icommands/commit/1ecd590eb9774584560f6f1ffc7e45d2254eb54b

File: src/itree.cpp
Comment (d-w-moore) [reply src/itree.cpp:42]: I took out the extra line.  However, a thing occurred to me: the <cassert> remains in 4-2-stable.  It won't hurt anything obvs., but these differences  that get introduced because of late-breaking review requests after one branch has already been merged seem ... problematic, if we desire 100% consistency.  

In terms of their effects in the code, of course, both of these branch differences are completely harmless. But I'll leave it up to others whether to apply the deletion of <cassert> and the blank line in the final product, and then also (obvs.) to whether to apply same by re-cherrypicking to 4-3-stable.

Comment (korydraughn) [reply src/itree.cpp:42]: You can include those last minute changes as a separate commit and then everything is aligned across branches.

Then all you have to do is cherry-pick the additional commit to the other branches.

Comment (d-w-moore) [reply src/itree.cpp:42]: Seems good, it just won't have a prefixed issue number in the title of the(new)4-2- stable pull request. Ok

Comment (korydraughn) [reply src/itree.cpp:42]: Sure. As long as the additional commit has an issue number, it is fine.

Comment (alanking) [reply src/itree.cpp:42]: This has been addressed, so I will resolve this. Thanks

**Review by alanking — COMMENTED** [review]
Please just squash to one commit and add #s

**Comment by d-w-moore** [comment]
'#' added.

**Review by alanking — APPROVED** [review]

**Pull request merged**
